web_ckeditor4/static/src/js/web_ckeditor4.js

```diff
@@ -105,59 +99,70 @@ openerp.web_ckeditor4 = function(instance)
         return ckeditor_writer.getHtml();
     };
 
-    default_ckeditor_filter = new CKEDITOR.filter(
+    var default_ckeditor_filter = new CKEDITOR.filter(
+        {
+            '*':
             {
-                '*':
-                {
-                    attributes: 'href,src,style,alt,width,height,dir',
-                    styles: '*',
-                    classes: '*',
-                },
-                'html head title meta style body p div span a h1 h2 h3 h4 h5 img br hr table tr th td ul ol li dd dt strong pre b i': true,
-            });
-    default_ckeditor_writer = new CKEDITOR.htmlParser.basicWriter();
-
-    instance.web_ckeditor4.FieldCKEditor4 = instance.web.form.FieldText.extend({
-        ckeditor_config: {
-            removePlugins: 'iframe,flash,forms,smiley,pagebreak,stylescombo',
-            filebrowserImageUploadUrl: 'dummy',
-            extraPlugins: 'filebrowser',
-            // this is '#39' per default which screws up single quoted text in ${}
-            entities_additional: '',
+                attributes: 'href,src,style,alt,width,height,dir',
+                styles: '*',
+                classes: '*',
+            },
+            'html head title meta style body p div span a h1 h2 h3 h4 h5 img br hr table tr th td ul ol li dd dt strong pre b i': true,
+        });
+    var default_ckeditor_writer = new CKEDITOR.htmlParser.basicWriter();
+    var FieldCKEditor4 = core.form_widget_registry.get('text').extend({
+        ckeditor_config: function () {
+          return {
+              removePlugins: this._getRemovePlugins(),
+              removeButtons: this._getRemoveButtons(),
+              filebrowserImageUploadUrl: 'dummy',
+              extraPlugins: 'filebrowser',
+              // this is '#39' per default which screws up single quoted text in ${}
+              entities_additional: ''
+          };
```

Comment:
Is it possible to add this line in the config ?
`allowedContent: true`
My problem is that ckeditor is removing classes from my html, but by adding this parameter we allow the attributes.

Reply:
@ecino i think that would be a bad idea, since it disables filtering completely. The odoo html field sanitizes the input so its not terrible, but i think we should keep it because it is stripped immediately and not only on save. This keeps the user from entering stuff which looks amazing only to have it removed when he saves.

Reply:
@ecino if you really want this, your use case is covered by `text_ckeditor4_raw` - but as @tschanzt points out, you'll also have to turn off filtering in the field to make this effective. and you have to trust your users a lot of course.

Reply:
Thanks for the tips, then I think you are right, we shouldn't add this.